store/iavlstore.go

```diff
@@ -102,7 +102,21 @@ func (st *iavlStore) LastCommitID() CommitID {
 	}
 }
 
-// VersionExists returns whether or not a given version is stored
+// Implements Committer.
+func (st *iavlStore) SetPruning(pruning string) {
+	switch pruning {
+	case "everything":
```

Comment:
Can `pruning` be an enum instead of a string, or is that hard to use with functional options?

Reply:
It could. Would just have to move this switch case into the baseapp.SetPruning function and have it translate to enum there, so that the Committer SetPruning function could take enums.

Reply:
An enum would be better I think; if the user specifies an invalid option we should fail.

Reply:
:+1: